pkg/topology/full/full.go

```diff
@@ -56,6 +56,8 @@ func (d *Driver) AddPeer(ctx context.Context, addr swarm.Address) error {
 		d.mtx.Unlock()
 		return nil
 	}
+
+	d.receivedPeers[addr.ByteString()] = struct{}{}
```

Comment:
@janos I moved this back here, to avoid having multiple add peer calls being handled. I think it's either this or a global lock across the whole function? I think this is a better approach, and we can remove the map entry on errors if needed. LMKWYT?

Reply:
I agree, it is ok.